src/libsync/networkjobs.h

```diff
@@ -399,6 +400,18 @@ class OWNCLOUDSYNC_EXPORT JsonApiJob : public AbstractNetworkJob
         _usePOST = usePOST;
     }
 
+    void usePUT(bool usePUT = true)
+    {
+        _usePUT = usePUT;
```

Comment:
Honestly, I think that code is a mess, but I don't want to touch it too much as there are just no tests and I'm afraid to break something. We could discuss about refactoring in another pr. What do you think?

Reply:
I fully disagree and think that @allexzander is true an enum should be used
there is no good reason to write code that can fail because we are afraid of breaking something
if tests are needed add them

Reply:
Fixed ;)